src/librustc_typeck/diagnostics.rs

```diff
@@ -3567,6 +3630,42 @@ fn together_we_will_rule_the_galaxy(son: &A<i32>) {} // Ok!
 ```
 "##,
 
+E0399: r##"
+You implemented a trait, overriding one or more of it's associated types but did
+not reimplement it's default methods. Example of erroneous code:
+
+```compile_fail,E0399
+#![feature(associated_type_defaults)]
+
+pub trait Foo {
+    type Assoc = u8;
+    fn bar(&self) {}
+}
+
+impl Foo for i32 {
+    // error - the following trait items need to be reimplemented as
+    //         `Assoc` was overridden: `bar`
+    type Assoc = i32;
+}
+```
+
+Add an implementation for the default methods:
```

Comment:
Maybe "To fix this, add an implementation for each default method from the trait:"?